uefi-mkconfig

```diff
@@ -74,13 +74,24 @@ add_uefi_entry () (
 		ewarn "No initramfs found for \"$partition_mount$efi_file_path\"."
 	fi
 
+	# Find disk name and gpt partition id
+	for line in $(lsblk -lno NAME); do
+        	if [[ "$partition" == *"$line"* ]] && [[ "$partition" != "$line" ]]; then
+                	local disk
+                	local partition_id
+                	disk="$line"
+                	partition_id="${partition//$line}"
+                	partition_id="${partition_id//p}"
+        	fi
+	done
+
```

Comment:
Why do we need to parse the output of lsblk again? `$partition` is already of the form `nvme0n[0-9]p[0-9]` or `sd[a-z][0-9]`:
```
* Creating UEFI entry "0200" for "/efi/EFI/Linux/linux-6.9.9-gentoo-dist.efi" found on "nvme0n1p1"
```
So we can simply separate this string into two parts `nvme0n1` and `p1` (and then drop the `p`) or `sda` and `1`).

Reply:
I suggest:
```
disk=${partition%[0-9]}
if [[ disk == nvme* ]]; then
      disk=${disk%p} # strip p only if nvme because sdp may exist
      part_num=${partition#nvme[0-9]n[0-9]p}
else
      part_num=${partition#sd[a-z]}
fi
```

Reply:
> Current code will break if the ESP is on /dev/sdp

This is not true. If we have partition sdp1 for example, `sdp` will be removed before getting to the code removing `p`.

Indeed I can avoid lablk again with better placement.



> I suggest:
> 
> ```
> disk=${partition%[0-9]}
> if [[ ${disk} == nvme* ]]; then
>       disk=${disk%p} # strip p only if nvme because sdp may exist
>       part_num=${partition#nvme[0-9]n[0-9]p}
> else
>       part_num=${partition#sd[a-z]}
> fi
> ```
> 
> It's shorter and avoids a for loop and calling lsblk again.

Problem with this is that if user uses some exotic block device other than nvme or sda (vda, mmcblk...)

Reply:
> Problem with this is that if user uses some exotic block device other than nvme or sda (vda, mmcblk...)

Is that supported by UEFI? We only have to worry about block devices in UEFI systems

> This is even more important since block device name scheme which linux will use can be customized.

Hmm, so it is not always sd[a-z] for sata disks?

Reply:
> So yes for loop isn't the best but it allows us to not hardcode block device names. 

Alright, dug a bit more into this and seems you are right. I'll merge your 2.2 PR.

Reply:
Thanks!